src/qrules/__init__.py

```diff
@@ -207,7 +213,7 @@ def check_edge_qn_conservation() -> set[frozenset[str]]:
         InteractionProperties(l_magnitude=l_magnitude, s_magnitude=s_magnitude)
         for l_magnitude, s_magnitude in product(
             _int_domain(0, max_angular_momentum),
-            _halves_domain(0, max_spin_magnitude),
+            _halves_domain(Fraction(0), Fraction(max_spin_magnitude)),
```

Comment:
I wonder if `_halves_domain()` can be rewritten in such a way that we can just write the following. I know this is an internal function, but for readability this syntax is better.
```suggestion
            _halves_domain(0, max_spin_magnitude),
```

Reply:
`_halves_domain` now just converts internally to `Fraction`. I will look where else `Fraction` is expected as user input and change it such that the user can just use `float`/`int` and convert at the first layer so that internally only `Fraction` is used.